http/codegen/openapi/json_schema.go

```diff
@@ -456,12 +458,26 @@ func buildAttributeSchema(api *design.APIExpr, s *Schema, at *design.AttributeEx
 		s.Maximum = val.Maximum
 	}
 	if val.MinLength != nil {
-		s.MinLength = val.MinLength
+		if _, ok := at.Type.(*design.Array); ok {
+			s.MinItems = val.MinLength
+		} else {
+			s.MinLength = val.MinLength
+		}
 	}
 	if val.MaxLength != nil {
-		s.MaxLength = val.MaxLength
+		if _, ok := at.Type.(*design.Array); ok {
+			s.MaxItems = val.MaxLength
+		} else {
+			s.MaxLength = val.MaxLength
+		}
 	}
 	s.Required = val.Required
+}
+
+// AttributeTypeSchema produces the JSON schema corresponding to the given attribute.
+func AttributeTypeSchema(api *design.APIExpr, at *design.AttributeExpr) *Schema {
```

Comment:
Is there a reason not to change `buildAttributeSchema` to `AttributeSchema` instead of introducing a new function? Or probably call `buildAttributeSchema` inside `AttributeSchema`? From what I see, both these functions do pretty much the same thing. `buildAttributeSchema` adds more fields like `Example`, `DefaultValue`, `Description`. Are they not desired?

Reply:
Sounds good! Thanks.